snuba/clusters/cluster.py

```diff
@@ -374,16 +376,37 @@ def __get_cluster_nodes(self, cluster_name: str) -> Sequence[ClickhouseNode]:
     if (s not in DEV_STORAGE_SETS or settings.ENABLE_DEV_FEATURES)
 }
 
-# Map all storages to clusters via storage sets
-_STORAGE_SET_CLUSTER_MAP = {
-    storage_set: cluster
-    for cluster in CLUSTERS
-    for storage_set in cluster.get_storage_set_keys()
-}
+_STORAGE_SET_CLUSTER_MAP: Optional[Dict[StorageSetKey, ClickhouseCluster]] = None
+
+
+def _get_storage_set_cluster_map() -> Dict[StorageSetKey, ClickhouseCluster]:
+    global _STORAGE_SET_CLUSTER_MAP
+    # Map all storages to clusters via storage sets
+    if _STORAGE_SET_CLUSTER_MAP is None:
+        _STORAGE_SET_CLUSTER_MAP = {
+            storage_set: cluster
+            for cluster in CLUSTERS
+            for storage_set in cluster.get_storage_set_keys()
+        }
+    return _STORAGE_SET_CLUSTER_MAP
```

Comment:
Why is this change needed? Wouldn't it be better to just mark the invalid cluster so that all the other valid clusters are on the map rather than not populate the map at all.